src/world/conquest_system.cpp

```diff
@@ -20,3 +20,282 @@ along with this program.  If not, see http://www.gnu.org/licenses/
+    const std::size_t dataLen      = headerLength + sizeof(bool) + sizeof(size_t) + sizeof(influence_t) * influences.size();
+    const uint8*      data         = new uint8[dataLen];
+
+    // Regional event type + conquest msg type
+    ref<uint8>((uint8*)data, 0) = REGIONAL_EVT_MSG_CONQUEST;
+    ref<uint8>((uint8*)data, 1) = CONQUEST_WORLD2MAP_INFLUENCE_POINTS;
+    ref<uint8>((uint8*)data, 2) = shouldUpdateZones;
+
+    // Influences controls array
+    ref<uint32>((uint8*)data, 3) = influences.size();
+    for (int i = 0; i < influences.size(); i++)
+    {
+        // Everything is offset by i*size of region control struct + headerLength
+        const std::size_t start              = headerLength + sizeof(bool) + sizeof(size_t) + i * sizeof(influence_t);
+        ref<uint16>((uint8*)data, start)     = influences[i].sandoria_influence;
+        ref<uint16>((uint8*)data, start + 2) = influences[i].bastok_influence;
+        ref<uint16>((uint8*)data, start + 4) = influences[i].windurst_influence;
+        ref<uint16>((uint8*)data, start + 6) = influences[i].beastmen_influence;
+    }
+
+    // 3- Create ZMQ Message and queue it
+    zmq::message_t dataMsg = zmq::message_t(dataLen);
+    memcpy(dataMsg.data(), data, dataLen);
+    if (ipp == 0)
+    {
+        queue_message_broadcast(MSG_WORLD2MAP_REGIONAL_EVENT, &dataMsg);
+    }
+    else
+    {
+        queue_message(ipp, MSG_WORLD2MAP_REGIONAL_EVENT, &dataMsg);
+    }
+}
+
+void ConquestSystem::sendRegionControlsMsg(CONQUESTMSGTYPE msgType, uint64 ipp)
+{
+    // 2- Serialize regional controls with the following schema:
+    // - REGIONALMSGTYPE
+    // - CONQUESTMSGTYPE
+    // - region controls array size
+    // - For N elements we have:
+    //      - current control (uint8)
+    //      - prev control (uint8)
+    auto regionControls = getRegionControls();
+
+    // Base length is the type + subtype + region control size
+    const std::size_t headerLength = 2 * sizeof(uint8);
+    const std::size_t dataLen      = headerLength + sizeof(region_control_t) * regionControls.size();
+    const uint8*      data         = new uint8[dataLen];
+
+    // Regional event type + conquest msg type
+    ref<uint8>((uint8*)data, 0) = REGIONAL_EVT_MSG_CONQUEST;
+    ref<uint8>((uint8*)data, 1) = msgType;
+
+    // Region controls array
+    ref<uint32>((uint8*)data, 2) = regionControls.size();
+    for (int i = 0; i < regionControls.size(); i++)
+    {
+        // Everything is offset by i*size of region control struct + headerLength + size of size_t
+        const std::size_t offset             = headerLength + sizeof(size_t) + sizeof(region_control_t) * i;
+        ref<uint8>((uint8*)data, offset)     = regionControls[i].current;
+        ref<uint8>((uint8*)data, offset + 1) = regionControls[i].prev;
+    }
+
+    // 3- Create ZMQ Message and queue it
+    zmq::message_t dataMsg = zmq::message_t(dataLen);
+    memcpy(dataMsg.data(), data, dataLen);
+
+    if (ipp == 0)
```

Comment:
I wonder if this is dangerous or not, because I think we use 0 as a designator for "Zone is turned off" in a lot of places?

Reply:
Alternative would be create a different method, or maybe a boolean flag for broadcast?

Reply:
If you're trying to use a magic number to send things to the world server, I'd probably go for 0xFFFF

Reply:
👍